README.md

```diff
@@ -15,8 +15,26 @@ This crate contains several layers:
    
 Rafx also provides tools for building shaders and packing assets.
 
-Rafx supports most mainstream platforms via `vulkan` and `metal` backends. Proprietary platforms can be supported by
-adding an additional backend.
+### Current Support:
+
+Rafx supports most mainstream platforms. Proprietary platforms can be supported by adding an additional backend.
```

Comment:
The changes look fine -- there's thin wrappers around OpenGL functions and lots of code to implement `rafx-api` or `rafx-framework` functions by matching on the backend and delegating to the new OpenGL code. There's a lot of code added with this branch but this sure looks like the code I would need to write to abstract away an entire rendering backend.

I think any more targeted feedback than that will require dedicated testing and usage with OpenGL projects to understand where this abstraction leaks / has perf issues / doesn't allow GL functionality that it ought to / etc.

There's nothing here architecturally that makes me worry about handling some hypothetical future issue though, so I think it makes sense to take a few days to clean it up, remove commented out code, address any minor TODOs that are low-hanging fruit, and then hit merge.

Reply:
Yeah, I think if I work on a GL ES 3.0 backend (that can work with the full-sized demo) it will uncover improvements that can be ported back to the 2.0. But I didn't want to start on a 3.0 backend without getting the 2.0 backend in first. :)

Reply:
(As an example, missing support for cube maps would naturally come as a result of getting an ES 3.0 backend working with the demo.)